strapi/strapi
# Api token v2/backend refresh lifespan

Draft until feature is approved

### What does it do?

Adds an admin api endpoint at `POST api-tokens/:id/refresh` to refresh a token.
Refresh means to set the current `token.expiresAt = Date.now() + lifespan`

### Why is it needed?

It is needed to extend the life of a token.

It must be an independent action from `update` or else the token would be refreshed every time it is edited, which we do not want. `update` can change the lifespan of an existing token, but another call to `refresh` must be done to extend the `expiresAt` date to use the new `lifespan`.

### How to test it?

- Create a token with a `lifespan` and note the `expiresAt` date returned
- Get the token and see that the `expiresAt` value is the new value
- Call the refresh endpoint with the token and see the `expiresAt` value has increased by `lifespan`
- Get the token and see that the `expiresAt` value is the new value

### Related issue(s)/PR(s)
